go/common/viewingkey/viewing_key.go

```diff
@@ -274,3 +310,18 @@ func CheckEIP712Signature(userID string, signature []byte, chainID int64) (*geth
 	}
 	return nil, errors.New("EIP 712 signature verification failed")
 }
+
+// CheckIfSignatureIsValidAndMatchesAddress checks if the signature is valid for the encryption token and matches the expected address
+func CheckIfSignatureIsValidAndMatchesAddress(encryptionToken string, signature []byte, chainID int64, expectedAddress *gethcommon.Address) bool {
```

Comment:
this approach is not great.
Let's add a "type int" field in the "RPCSignedViewingKey".
The type will be an iota of either "EIP712" or "Personal".

Based on that type, you choose the signature verification method